apps/src/studio/studio.js

```diff
@@ -347,10 +347,9 @@ var drawMap = function () {
     }
   }
 
+  var goalOverride = utils.valueOr(level.goalOverride, {});
   var numFrames = 1;
-  if (level.goalOverride && 
-      level.goalOverride.goalAnimation && 
-      skin.animatedGoalFrames) {
+  if (goalOverride.goalAnimation && skin.animatedGoalFrames) {
```

Comment:
Note: This should be a nonfunctional change, I've just extracted the null-check for goalOverride throughout `drawMap`.